### PR TITLE
Refactor OnNewHead to return a channel

### PR DESCRIPTION
**Pull request by EasterTheBunny**
The behavior of `OnNewHead` resulted in a reference back to the calling service by passing a function to the subscription. Converting to a channel eliminates the outside reference to service and only creates references from service to outside entities.

### CODE REVIEW

File: pkg/chain/evmregistry.go [hunk 1]
```diff
@@ -17,18 +17,7 @@ import (
 	"github.com/smartcontractkit/ocr2keepers/pkg/types"
 )
 
-const (
-	ActiveUpkeepIDBatchSize int64  = 10000
-	separator               string = "|"
-)
-
```
Comment (infiloop2) [inline pkg/chain/evmregistry.go:24]: is this strange strictly needed? it looks safe as just a move, but trying to minimise the changes in bugfix

File: pkg/chain/evmclient.go [hunk 2]
```diff
@@ -61,23 +67,27 @@ func (ec *evmClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) er
...
-			return ctx.Err()
+			return
 		case <-sub.Err():
 			if sub, err = ec.SubscribeNewHead(ctx, ch); err != nil {
-				return err
+				return
 			}
 		case head := <-ch:
-			cb(types.BlockKey(head.Number.String()))
+			ec.chHead <- types.BlockKey(head.Number.String())
```
Comment (infiloop2) [inline pkg/chain/evmclient.go:90]: will this be a blocking call if chHead is full?

File: pkg/chain/evmclient.go [hunk 3]
```diff
@@ -19,17 +19,23 @@ import (
...
 	rpcClient *rpc.Client
 	batchSize int
 }
 
 // NewEVMClient is the constructor of evmClient
 func NewEVMClient(client *rpc.Client, batchSize int) types.EVMClient {
-	return &evmClient{
+	c := &evmClient{
 		Client:    ethclient.NewClient(client),
+		chHead:    make(chan types.BlockKey, 1000),
```
Comment (infiloop2) [inline pkg/chain/evmclient.go:31]: any reason to not have it 1 length as we had before

Comment (EasterTheBunny) [reply pkg/chain/evmclient.go:31]: Same as above

File: pkg/chain/evmregistry.go (hunk 1 above)
Comment (EasterTheBunny) [reply pkg/chain/evmregistry.go:24]: No. But like the client, the evm registry isn't used by the chainlink node. It is only used by the simulator.

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**